### PR TITLE
NIFI-3031 - Multi-Statement Script support for PutHiveQL Processor

### PR DESCRIPTION
**Pull request by dstreev**
Thank you for submitting a contribution to Apache NiFi.

In order to streamline the review of the contribution we ask you
to ensure the following steps have been taken:

### For all changes:
- [x] Is there a JIRA ticket associated with this PR? Is it referenced 
     in the commit message?

- [x] Does your PR title start with NIFI-XXXX where XXXX is the JIRA number you are trying to resolve? Pay particular attention to the hyphen "-" character.

- [x] Has your PR been rebased against the latest commit within the target branch (typically master)?

- [ ] Is your initial contribution a single, squashed commit?

### For code changes:
- [x] Have you ensured that the full suite of tests is executed via mvn -Pcontrib-check clean install at the root nifi folder?
- [x] Have you written or updated unit tests to verify your changes?
- [ ] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](http://www.apache.org/legal/resolved.html#category-a)? 
- [ ] If applicable, have you updated the LICENSE file, including the main LICENSE file under nifi-assembly?
- [ ] If applicable, have you updated the NOTICE file, including the main NOTICE file found under nifi-assembly?
- [ ] If adding new Properties, have you added .displayName in addition to .name (programmatic access) for each of the new properties?

### For documentation related changes:
- [ ] Have you ensured that format looks appropriate for the output in which it is rendered?

### Note:
Please ensure that once the PR is submitted, you check travis-ci for build issues and submit an update to your PR as soon as possible.

### CODE REVIEW

**Comment by mattyb149** [comment]
Reviewing...

**Comment by patricker** [comment]
@dstreev I had been preparing to write this very enhancement to support Multi-Statement scripts for several weeks; thanks for taking care of it.

I didn't pay close attention to your changes for CSV, but the parts on multi-statement look good.  I will have to pull it down and give it a run.

File: nifi-nar-bundles/nifi-hive-bundle/nifi-hive-processors/src/main/java/org/apache/nifi/processors/hive/SelectHiveQL.java [hunk 1]
```diff
@@ -135,42 +197,100 @@
...
         final StopWatch stopWatch = new StopWatch(true);
+        final boolean header = context.getProperty(HIVEQL_CSV_HEADER).asBoolean();
+        final String altHeader = context.getProperty(HIVEQL_CSV_ALT_HEADER).evaluateAttributeExpressions(fileToProcess).getValue();
+        final String delimiter = context.getProperty(HIVEQL_CSV_DELIMITER).evaluateAttributeExpressions(fileToProcess).getValue();
+        final boolean quote = context.getProperty(HIVEQL_CSV_QUOTE).asBoolean();
+        final boolean escape = context.getProperty(HIVEQL_CSV_HEADER).asBoolean();
 
         try (final Connection con = dbcpService.getConnection();
-             final Statement st = con.createStatement()) {
+             final Statement st = ( flowbased ? con.prepareStatement(selectQuery): con.createStatement())
```
Comment (mattyb149) [inline nifi-nar-bundles/nifi-hive-bundle/nifi-hive-processors/src/main/java/org/apache/nifi/processors/hive/SelectHiveQL.java]: Isn't it possible to specify a parameterized query in the Select Query property, expecting that each flow file has the appropriate attributes set? I'm wondering if there's a better check to know when to call prepareStatement vs createStatement.

File: nifi-nar-bundles/nifi-hive-bundle/nifi-hive-processors/src/main/java/org/apache/nifi/processors/hive/SelectHiveQL.java [hunk 2]
```diff
@@ -90,11 +98,59 @@
...
+            .required(true)
+            .allowableValues("true", "false")
+            .defaultValue("true")
+            .addValidator(StandardValidators.BOOLEAN_VALIDATOR)
+            .build();
+
+    public static final PropertyDescriptor HIVEQL_CSV_ALT_HEADER = new PropertyDescriptor.Builder()
+            .name("csv-alt-header")
+            .displayName("Alternate CSV Header")
+            .description("Comma separated list of header fields")
```
Comment (dstreev) [inline nifi-nar-bundles/nifi-hive-bundle/nifi-hive-processors/src/main/java/org/apache/nifi/processors/hive/SelectHiveQL.java]: I've had a few instances where the declared fieldname, say '_date' or 'date_' , doesn't work well in the header.  So you want the option to replace it with 'date'.

Comment (mattyb149) [inline nifi-nar-bundles/nifi-hive-bundle/nifi-hive-processors/src/main/java/org/apache/nifi/processors/hive/SelectHiveQL.java]: Makes sense, I'm thinking we should make that kind of point right in the documentation

Comment (dstreev) [inline nifi-nar-bundles/nifi-hive-bundle/nifi-hive-processors/src/main/java/org/apache/nifi/processors/hive/SelectHiveQL.java]: The comment above removed the 'underscore'.

IE: \_date or date\_

**Comment by diser55** [comment]
@dstreev - multiple lines support was added for PutHiveQL only, not SelectHiveQL, right?

**Pull request closed without merging**